src/hooks/useWizardForm/useWizardForm.tsx

```diff
@@ -136,7 +136,18 @@ const useWizardForm = ({
   }
 
   const handleGoToNextQuestion = async () => {
-    await updateQuestionVisibility().finally(() => {
+    await updateQuestionVisibility().then((updatedQuestions) => {
+      // check if there are no new visible questions after evaluating rules
+      const doNextQuestionExist = current !== updatedQuestions.length - 1
+      if (!doNextQuestionExist) {
+        // check if there are any errors
+        const hasErrors = handleCheckForErrors()
+        // if there are no errors, submit the form
+        if (!hasErrors) {
+          formMethods.handleSubmit(handleConvertAndSubmitForm)()
+        }
+        return
+      }
```

Comment:
```suggestion
      // check if there are no new visible questions after evaluating rules
      const isLastVisibleQuestion = current === updatedQuestions.length - 1
      if (isLastVisibleQuestion) {
        // check if there are any errors
        const hasErrors = handleCheckForErrors()
        // if there are no errors, submit the form
        if (!hasErrors) {
          formMethods.handleSubmit(handleConvertAndSubmitForm)()
        }
        return
      }
```

Reply:
Is there a risk that this change introduces a regression bug if the `updateQuestionVisibility` function throws an error ? If I'm not mistaken before the change we use `finally`, which will be executed even if `updateQuestionVisibility` throws, whereas now any error will prevent the code below to be executed.

Reply:
hadn't considered it that way. refactored to:

```
  const handleGoToNextQuestion = async () => {
    const hasErrors = handleCheckForErrors()
    if (!hasErrors) {
      try {
        const updatedQuestions = await updateQuestionVisibility()
        const isLastVisibleQuestion = current === updatedQuestions.length - 1
        if (isLastVisibleQuestion) {
          formMethods.handleSubmit(handleConvertAndSubmitForm)()
          return
        }
      } finally {
        setCurrent(current + 1)
      }
    }
    if (current === -1) {
      setCurrent(current + 1)
    }
  }
  ```